includes/exceptions.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(OCOS_NO_EXCEPTIONS) || defined(OCOS_CONTAIN_EXCEPTIONS)
+#if defined(__ANDROID__)
+#include <android/log.h>
+#else
+#include <iostream>
+#endif
+#endif
+
+#include <stdexcept>
+
+#include "onnxruntime_c_api.h"
+
+namespace OrtW {
+
+// All C++ methods that can fail will throw an exception of this type
+struct Exception : std::exception {
+  Exception(std::string&& string, OrtErrorCode code) : message_{std::move(string)}, code_{code} {}
+
+  OrtErrorCode GetOrtErrorCode() const { return code_; }
+  const char* what() const noexcept override { return message_.c_str(); }
+
+ private:
+  std::string message_;
+  OrtErrorCode code_;
+};
+
+#if defined(OCOS_NO_EXCEPTIONS) || defined(OCOS_CONTAIN_EXCEPTIONS)
+inline void PrintFinalMessage(const char* file, int line, const char* msg) {
+#if defined(__ANDROID__)
+  __android_log_print(ANDROID_LOG_ERROR, "onnxruntime-extensions", "Exception in %s line %d: %s", file, line, msg);
+#else
+  std::cerr << "Exception in " << file << " line " << line << ": " << msg << std::endl;
+#endif
+}
+#endif
+
+#ifdef OCOS_NO_EXCEPTIONS
+#define ORTX_CXX_API_THROW(string, code)                                         \
```

Comment:
It would be good to make all the naming/namespaces consistent in a separate PR. We have `OrtW` (not sure what that means), ORTX_, OCOS_ and various other things in places. We should agree on a macro prefix and namespace and update everything to use those. 

I think that's a separate PR though as it will touch a lot of files.

Reply:
ok, sounds good